packages/helpers/src/modules/assetValue.ts

```diff
@@ -103,16 +110,29 @@ export class AssetValue extends BigIntArithmetics {
     return this.eqAsset(assetValue) && this.eqValue(assetValue);
   }
 
-  // THOR.RUNE
-  // THOR.ETH.ETH
-  // ETH.THOR-0x1234567890
   static fromUrl(urlAsset: string, value: NumberPrimitives = 0) {
-    const [chain, ticker, symbol] = urlAsset.split(".");
-    if (!(chain && ticker)) {
+    const firstDotIndex = urlAsset.indexOf(".");
+
+    if (firstDotIndex === -1) {
       throw new SwapKitError({ errorKey: "helpers_invalid_asset_url", info: { urlAsset } });
     }
 
-    const asset = chain === Chain.THORChain && symbol ? `${chain}.${ticker}/${symbol}` : urlAsset;
+    const chain = urlAsset.substring(0, firstDotIndex);
+    const rest = urlAsset.substring(firstDotIndex + 1);
+    let asset: string;
+
+    // Check for trade asset encoding '..'
+    if (rest.includes("..")) {
+      asset = `${chain}.${rest.replace("..", "~")}`;
+    }
+    // Check for THORChain synth encoding (e.g., THOR.AVAX.AVAX)
+    else if ([Chain.THORChain, Chain.Maya].includes(chain as Chain) && rest.includes(".")) {
+      asset = `${chain}.${rest.replace(".", "/")}`;
+    }
+    // Default case: decode our '__' for NEAR and others
+    else {
+      asset = `${chain}.${rest.replace(/__/g, ".")}`;
+    }
```

Comment:
That part looks like a perfect `match` usage from ts-pattern (clear pattern matching for behaviour there.

Additionally - I don't feel real difference between `substring` and `slice` (unless you mess up with putting parameters in wrong order) but overall we "prefer" usage of slice in code (like few substring vs +80 slices)

Performance is negligible